web_src/js/features/repo-settings-branches.ts

```diff
@@ -0,0 +1,30 @@
+import {createSortable} from '../modules/sortable.ts';
+import {POST} from '../modules/fetch.ts';
+
+export function initRepoBranchesSettings() {
+  const protectedBranchesList = document.querySelector(
+    '#protected-branches-list',
+  );
+  if (!protectedBranchesList) return;
+
+  createSortable(protectedBranchesList, {
+    handle: '.drag-handle',
+    animation: 150,
+    onEnd: async () => { // eslint-disable-line @typescript-eslint/no-misused-promises
+      const newOrder = Array.from(protectedBranchesList.children, (item) => {
+        const id = item.getAttribute('data-id');
+        return id ? parseInt(id) : NaN;
+      }).filter((id) => !Number.isNaN(id));
```

Comment:
Why the filter is needed? If an item doesn't have `data-id`, isn't is just a bug?

Reply:
It's not needed but I added it just as savety guard as it does not hurt -> defensive programming (e.g. I expect users to mess up there custom templates)

Reply:
No, that's not defensive programming, that's unclear logic.

You can't do use `protectedBranchesList.children` directly here, but need to query correct items first.

Reply:
`queryElemChildren(protectedBranchesList, '.item[data-id]')`

Then no need the `filter` or `isNaN` trick